cnf-certification-test/daemonset/daemonset.go

```diff
@@ -0,0 +1,231 @@
+					},
+					Volumes: []v1core.Volume{
+						{
+							Name: "host",
+							VolumeSource: v1core.VolumeSource{
+								HostPath: &v1core.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	fmt.Printf("Deleting daemon set %s\n", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	if err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}); err != nil {
+		logrus.Infof("The daemonset (%d) deletion is unsuccessful due to %+v", daemonSetName, err.Error())
+	}
+	doneCleanUp := false
+	for start := time.Now(); !doneCleanUp && time.Since(start) < timeout; {
+		client := clientsholder.GetClientsHolder().K8sClient
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+
+		if len(pods.Items) == 0 {
+			doneCleanUp = true
+			break
+		}
+		time.Sleep(time.Duration(timeout.Minutes()))
+	}
+
+	fmt.Printf("Successfully cleaned up daemon set %s\n", daemonSetName)
+	return nil
+}
+
+// Check if the daemon set exists
+func doesDaemonSetExist(daemonSetName, namespace string) bool {
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	_, err := client.DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Println("Error ocurred for" + err.Error())
+	}
+	// If the error is not found, that means the daemon set exists
+	return err == nil
+}
+func CreateDaemonSet(daemonSetName, namespace, containerName, imageWithVersion string, timeout time.Duration) (*v1core.PodList, error) {
+	rebootDaemonSet := CreateDaemonSetsTemplate(daemonSetName, namespace, containerName, imageWithVersion)
+	if doesDaemonSetExist(daemonSetName, namespace) {
+		err := DeleteDaemonSet(daemonSetName, namespace)
+		if err != nil {
+			logrus.Debug("Failed to delete L2discovery daemonset because: %s", err)
+		}
+	}
+
+	fmt.Printf("Creating daemon set %s\n", daemonSetName)
+	client := clientsholder.GetClientsHolder().K8sClient
+	_, err := client.AppsV1().DaemonSets(namespace).Create(context.TODO(), rebootDaemonSet, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	WaitDaemonsetReady(namespace, daemonSetName, timeout)
+
+	fmt.Println("Deamon set is ready")
+
+	var ptpPods *v1core.PodList
+	ptpPods, err = client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+	if err != nil {
+		return ptpPods, err
+	}
+	fmt.Printf("Successfully created daemon set %s\n", daemonSetName)
+	return ptpPods, nil
+}
+
+func partnerRepoDaemonset(daemonSetName, namespace, containerName, imageWithVersion string) {
```

Comment:
This function doesn't have a return value or handle potential errors.

Reply:
I fixed it.